src/api/records.tsx

```diff
@@ -80,6 +85,34 @@ export const useRecordsPaginated = (): UseQueryResult<
   );
 };
 
+// TODO: should we integrate this with useRecordsPaginated? and have options to
+// pass the recordRow select function and which of the query params to apply?
```

Comment:
So have approprate `useQueryResult` options for fetching pages of records (for use in Data tab) and options for fetching records by date range (for use in Plot tab)? Yeah I'm down for that. We already do it in `channels.tsx` and I quite like how it looks. Feel free to spread out the code into separate files if it makes sense to do that. For example, we can keep `records.tsx` purely for fetching data and other functions in other files in `src/api` for curating the options and calls to the `useRecords` function.

Reply:
I probably will separate them then - it also makes it easier to mock just the data fetching functions whilst leaving the react-query functionality if they're in separate files (see `dg-download`).

Reply:
I'm happy with that. Make a Jira issue if necessary so we don't lose track :)